app/controllers/account_invitations_controller.rb

```diff
@@ -18,8 +21,33 @@ def destroy
     end
   end
 
+  def accept_invitation
+    if params.key?(:invitation_id)
+      redirect_to root_path unless received_invitation && token_valid? && invitation_belongs_to_current_user?
```

Comment:
if `invitation_belongs_to_current_user?` then it's an error (no weird redirects), 404 would be fine.

also check that it was not used before. both an invalid or used invitation should give 404 error.